libs/hdf-converters/src/veracode-mapper.ts

```diff
@@ -0,0 +1,552 @@
+/* eslint-disable prettier/prettier */
+import { integer } from 'aws-sdk/clients/cloudfront';
+import parser from 'fast-xml-parser';
+import {ExecJSON} from 'inspecjs';
+import _ from 'lodash';
+import {version as HeimdallToolsVersion} from '../package.json';
+import {
+  BaseConverter,
+  ILookupPath,
+  MappedTransform
+} from './base-converter';
+import {CweNistMapping} from './mappings/CweNistMapping';
+const CWE_LENGTH = 'cwe.length'
+const SCA_VULNERABILITIES = 'detailedreport.software_composition_analysis.vulnerabilities['
+const REPORT_SEVERITY = 'detailedreport.severity['
+const SCA_CVES = 'detailedreport.software_composition_analysis.cves['
+const CATEGORY = '].category['
+const CWE_NIST_MAPPING = new CweNistMapping();
+const DEFAULT_NIST_TAG = ['SI-2', 'RA-5'];
+const IMPACT_MAPPING: Map<string, number> = new Map([
+  ['5', 0.9],
+  ['4', 0.7],
+  ['3', 0.5],
+  ['2', 0.3],
+  ['1', 0.1],
+  ['0', 0.0]
+]);
+
+function impactMapping(severity: number): number {
+  if (typeof severity === 'string' || typeof severity === 'number') {
+    return IMPACT_MAPPING.get(severity.toString().toLowerCase()) || 0.1;
+  } else {
+    return 0.1;
+  }
+}
+
+function nistTag(input: string): string[] {
+  const len = Number(`${_.get(input, CWE_LENGTH)}`)
+  const cwes = []
+  for (let index = 0; index < len; index++) {
+    cwes.push(`${_.get(input, 'cwe.cweid')}`)
+  }
+  return CWE_NIST_MAPPING.nistFilter(cwes, DEFAULT_NIST_TAG).concat(['Rev_4']);
+}
+
+function formatRecommendations(input: unknown): string {
+  const text = [];
+  if (_.has(input, 'recommendations.para')) {
+    if (`${_.get(input, 'recommendations.para.text')}` !== "undefined") {
+      text.push(
+        `${_.get(input, 'recommendations.para.text')}`
+      );
+    }
+    else {
+      const len = Number(`${_.get(input, 'recommendations.para.length')}`)
+      for (let index = 0; index < len; index++) {
+        const topush = `recommendations.para[${index}].text`
+        text.push(`${_.get(input, topush)}`)
+      }
+    }
+  }
+  if (_.has(input, 'recommendations.para.bulletitem')) {
```

Comment:
see above.  basically for loops can be annoying and a possible source of errors so i try to avoid them if possible

also is text guaranteed to be within bulletitem?  otherwise might wanna do a has check here too for that